low-level/framework/base/sspl_constants.py

```diff
@@ -116,22 +116,35 @@
 salt_uniq_attr_per_node = ['cluster_id']
 salt_uniq_passwd_per_node = ['RABBITMQINGRESSPROCESSOR', 'RABBITMQEGRESSPROCESSOR', 'LOGGINGPROCESSOR']
 
-try:
-    setup_info = subprocess.Popen(['sudo', '/usr/bin/provisioner', 'get_setup_info'],
-        stdout=subprocess.PIPE).communicate()[0].decode("utf-8").rstrip()
-    setup_info = ast.literal_eval(setup_info)
-    storage_type = setup_info['storage_type'].lower()
-    server_type = setup_info['server_type'].lower()
-    logger.info(f"Storage Type : '{storage_type}'")
-    logger.info(f"Server Type '{server_type}'")
+# Provisioner API Call Keys
+NODE_KEY = 'nodes'
+SERVER_PER_NODE_KEY = 'servers_per_node'
+STORAGE_TYPE_KEY = 'storage_type'
+SERVER_TYPE_KEY = 'server_type'
+STORAGE_TYPE = 'physical'
+SERVER_TYPE = 'rbod'
+
 
+try:
+    print('fetching the setup values:')
+    process = subprocess.Popen(['sudo', '/usr/local/bin/provisioner', 'get_setup_info'], \
+                shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    _result, _ret_code = process.communicate()[0],process.returncode
+    if _ret_code == 0:
+        setup_info = eval(_result.decode("utf-8").rstrip())
+        STORAGE_TYPE = setup_info[STORAGE_TYPE_KEY].lower()
+        SERVER_TYPE = setup_info[SERVER_TYPE_KEY].lower()
+    logger.info(f'Storage Type: {STORAGE_TYPE}, Server Type: {SERVER_TYPE}')
+    print(f'values: {STORAGE_TYPE} {SERVER_TYPE}')
```

Comment:
Why print and logger both?

Reply:
Because, It gets executed at init level and I think inside binary as well.